Monika After Story/game/styles.rpy

```diff
@@ -388,3 +390,148 @@ init -10 python in mas_ui:
             if data["return_value"] == data["true_value"]:
                 return selected_prompt
         return default_prompt
+
+    # Methods for twopane_scrollable_menu
+    def _twopane_menu_filter_events(ev, search_query, search_kws, only_pool, only_random, only_unseen, only_seen):
+        """
+        A method to use as the filter for events in the twopane menu
+
+        IN:
+            ev - event object
+            search_query - search query to filter by
+            search_kws - search_query splitted using spaces
+
+        OUT:
+            boolean whether or not the event pass the criteria
+        """
+        ev_label = ev.eventlabel.lower()
+        ev_prompt = ev.prompt.lower()
+        ev_prompt_kws = ev_prompt.split()
+
+        return (
+            ev_prompt != ev_label
+            and ev.unlocked
+            # and (ev.pool or ev.random)
+            and not ev.anyflags(store.EV_FLAG_HFNAS)
+            and ev.checkAffection(store.mas_curr_affection)
+            and (
+                (not only_pool or ev.pool)
+                and (not only_random or ev.random)
+                and (not only_unseen or ev.shown_count == 0)
+                and (not only_seen or ev.shown_count > 0)
+            )
+            and (
+                search_query == ev_prompt
+                or search_query == ev_label
+                or search_query in ev_prompt
+                or search_query in ev_label
+                or (ev.category and search_query in ev.category)
+                or all(
+                    [
+                        search_kw in ev_prompt
+                        or search_kw in ev_label
+                        or (ev.category and search_kw in ev.category)
+                        for search_kw in search_kws
+                    ]
+                )
+            )
+        )
+
+    def _twopane_menu_sort_events(ev, search_query, search_kws):
+        """
+        A method to use as the sortkey for events in the twopane menu.
+
+        IN:
+            ev - event object
+            search_query - search query to filter by
+            search_kws - search_query splitted using spaces
+
+        OUT:
+            list of booleans for sorting
+        """
+        ev_label = ev.eventlabel.lower()
+        ev_label_kws = ev_label.split("_")
+        ev_prompt = ev.prompt.lower()
+        ev_prompt_kws = ev_prompt.split()
+
+        rv = list()
+
+        rv.append(search_query == ev_prompt or search_query == ev_label)
+        rv.append(search_query in ev_prompt)
+        rv.append(search_query in ev_label)
```

Comment:
similar thing here with `==` and `in`, although I see it might not be functionally equivalent. let me know if the `==` checks are here for a reason.